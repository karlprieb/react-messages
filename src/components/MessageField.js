import React, { PureComponent } from 'react'
import PropTypes from 'prop-types'
import { IconSend } from '../icons'

import './messageField.css'

class MessageField extends PureComponent {
  constructor (props) {
    super(props)
    this.state = {
      value: ''
    }

    this.handleChange = this.handleChange.bind(this)
    this.handleSubmit = this.handleSubmit.bind(this)
  }

  handleChange (event) {
    this.setState({value: event.target.value})
  }

  handleSubmit (event) {
    event.preventDefault()
    const text = this.state.value
    if (text) {
      this.props.onSubmit(text)
      this.setState({value: ''})
    }
  }

  render () {
    return (
      <React.Fragment>
        <form className='message-form' onSubmit={this.handleSubmit}>
          <textarea className='message-textarea' value={this.state.value} onChange={this.handleChange} />
          <button className='message-send-button' type='submit'>
            <IconSend title='Send' />
          </button>
        </form>
      </React.Fragment>
    )
  }
}

MessageField.propTypes = {
  onSubmit: PropTypes.func.isRequired
}

export default MessageField
