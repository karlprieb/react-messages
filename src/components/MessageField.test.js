import React from 'react'
import MessageField from './MessageField'

const onSubmit = jest.fn()
const component = <MessageField onSubmit={onSubmit} />
const wrapper = shallow(component)

describe('Message textarea', () => {
  it('shoud render without error', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should clear textarea value on submit', () => {
    // Add some value to textarea
    wrapper.find('textarea').simulate('change', {target: {value: 'textarea message'}})
    expect(wrapper.find('textarea').getElement().props.value).toEqual('textarea message')

    // Submit the form to clear textarea
    wrapper.find('form').simulate('submit', { preventDefault () {} })
    expect(wrapper.find('textarea').getElement().props.value).toEqual('')
  })
})
