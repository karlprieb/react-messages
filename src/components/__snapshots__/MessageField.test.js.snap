// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message textarea shoud render without error 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <textarea
      onChange={[Function]}
      value=""
    />
    <button
      type="submit"
    >
      Send
    </button>
  </form>
</div>
`;
